dsl/skip.rb

```diff
@@ -0,0 +1,34 @@
+# typed: true
+# frozen_string_literal: true
+
+#: self as Roast::DSL::Workflow
+
+config do
+  cmd { print_all! }
+end
+
+execute do
+  cmd(:seconds) { "date +'%S'" }
+
+  cmd(:even) do
+    # `cmd!` output accessor bang method will raise an exception if the named cog did not complete successfully
+    seconds = cmd!(:seconds).out.strip.to_i
+    # Use the `skip!` method in a cog's input proc to conditionally skip this cog
+    # If `skip!` is called, the input proc will immediately terminate
+    skip! if seconds.odd?
+    "echo '#{seconds} is even'"
+  end
+
+  cmd(:odd) do
+    seconds = cmd!(:seconds).out.strip.to_i
+    skip! if seconds.even?
+    "echo '#{seconds} is odd'"
+  end
+
+  cmd do |my|
+    my.command = "echo"
+    # `cmd` output accessor non-bang method will return nil if the named cog did not run yet or did not complete successfully
+    my.args << "'even' cog ran" unless cmd(:even).nil?
```

Comment:
Instead of `nil?`, what about this:

```suggestion
    my.args << "'even' cog ran" unless cmd?(:even)
```

Reply:
Yes, definitely. Any attempt to reference an invalid cog should fail loudly, since that should never be something you want to do.

Reply:
done!